src/flake8_simplify/rules/ast_with.rs

```diff
@@ -1,22 +1,54 @@
 use rustpython_ast::{Stmt, StmtKind};
 
+use crate::ast::helpers::first_colon_range;
 use crate::ast::types::Range;
 use crate::checkers::ast::Checker;
 use crate::registry::Diagnostic;
 use crate::violations;
 
-/// SIM117
-pub fn multiple_with_statements(checker: &mut Checker, stmt: &Stmt) {
+fn find_nested_with(stmt: &Stmt) -> Option<&Stmt> {
```

Comment:
I think the return type here could be precised, since it will always return a `StmtKind::With` ... since rustpython_ast unfortunately doesn't define a struct for the fields of the `StmtKind::With` variant we'd have to define that struct ourselves ... but I think it would still be worth it for the additional clarity ... and then we could remove the weird looking `else { return }` in the `multiple_with_statements` function.